ADAL/src/request/ADAuthenticationRequest.m

```diff
@@ -268,4 +268,14 @@ - (NSUUID*)correlationId
     return _correlationId;
 }
 
+- (BOOL)takeUserInterationLock
+{
+    return OSAtomicCompareAndSwapInt( 0, 1, &sInteractionInProgress);
```

Comment:
It's possible to use a dispatch_semaphore as a lock. Just have it return immediately on the wait and check whether or not you got the semaphore to try and get the lock and signal to let go of the lock.